source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -41,6 +40,7 @@ FaultSettings::FaultSettings(const envoy::config::filter::http::fault::v2::HTTPF
 
   if (fault.has_delay()) {
     fixed_delay_percent_ = fault.delay().percent();
+    fixed_delay_percentage_ = fault.delay().percentage();
```

Comment:
do you have validtion set to ensure that this field is not null ?
Also, fold fixed_delay_percent_ into fixed_delay_percentage.. i.e. convert the old percent (if set) to new form..
The problem though is that you can't differentiate between a 0 percent and unset percent in hte old one as its uint32.. Sigh..

Reply:
So, we keep this as is with the validation brought in?

Reply:
I've addressed this in 6652866.